examples/pod_exec_terminal_size.rs

```diff
@@ -0,0 +1,132 @@
+            Some(Ok(_)) => {},
+            Some(Err(err)) => {
+                return Err(err.into());
+            }
+            None => break,
+        }
+    }
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::init();
+    let client = Client::try_default().await?;
+
+    let pods: Api<Pod> = Api::default_namespaced(client);
+    let p: Pod = serde_json::from_value(serde_json::json!({
+        "apiVersion": "v1",
+        "kind": "Pod",
+        "metadata": { "name": "example" },
+        "spec": {
+            "containers": [{
+                "name": "example",
+                "image": "alpine",
+                // Do nothing
+                "command": ["tail", "-f", "/dev/null"],
+            }],
+        }
+    }))?;
+    // Create pod if don't exist
+    pods.create(&PostParams::default(), &p).await?;
+
+    // Wait until the pod is running, otherwise we get 500 error.
+    let running = await_condition(pods.clone(), "example", is_pod_running());
+    let _ = tokio::time::timeout(std::time::Duration::from_secs(15), running).await?;
+
+    {
+        crossterm::terminal::enable_raw_mode()?;
+        let mut attached: AttachedProcess = pods
+            .exec(
+                "example",
+                vec!["sh"],
+                &AttachParams::default().stdin(true).tty(true).stderr(false),
+            )
+            .await?;
+
+        let mut stdin = tokio_util::io::ReaderStream::new(tokio::io::stdin());
+        let mut stdout = tokio::io::stdout();
+
+        let mut output = tokio_util::io::ReaderStream::new(attached.stdout().unwrap());
+        let mut input = attached.stdin().unwrap();
+
+        let term_tx = attached.terminal_size().unwrap();
+
+        let mut handle_terminal_size_handle = tokio::spawn(handle_terminal_size(term_tx));
+
+        loop {
+            select! {
+                message = stdin.next() => {
+                    match message {
+                        Some(Ok(message)) => {
+                            input.write(&message).await?;
+                        }
+                        _ => {
+                            break;
+                        },
+                    }
+                },
+                message = output.next() => {
+                    match message {
+                        Some(Ok(message)) => {
+                            stdout.write(&message).await?;
+                            stdout.flush().await?;
+                        },
+                        _ => {
+                            break
+                        },
+                    }
+                },
+                result = &mut handle_terminal_size_handle => {
+                    match result {
+                        Ok(_) => println!("End of terminal size stream"),
+                        Err(e) => println!("Error getting terminal size: {e:?}")
+                    }
+                },
+            };
+        }
```

Comment:
Maybe add a comment here about how this is emulating interactive terminal i/o (since it's the only example we have that does this).